File: frontend/src/types/manga.ts
export type ReadingStatus = 'plan_to_read' | 'reading' | 'dropped' | 'completed'

export type MangaCategory =
  | 'hot_blooded'
  | 'mystery'
  | 'adventure'
  | 'romance'
  | 'casual'
  | 'competition'
  | 'revenge'
  | 'slice_of_life'
  | 'other'

export interface CollectionItem {
  id: number
  mangaId: number
  title: string
  category: MangaCategory
  status: ReadingStatus
  currentVolume: number | null
  currentChapter: number | null
  rating: number | null
  lastReadAt: string
  createdAt: string
  updatedAt: string
}

export interface MangaSearchResult {
  id: number
  title: string
  category: MangaCategory
}

export interface MangaAdminItem {
  id: number
  title: string
  category: MangaCategory
  createdAt: string
  updatedAt: string
}

File: backend/app/model/manga.py
import enum
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, Enum, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.model.base import Base


class MangaCategory(str, enum.Enum):
    HOT_BLOODED = "hot_blooded"
    MYSTERY = "mystery"
    ADVENTURE = "adventure"
    ROMANCE = "romance"
    CASUAL = "casual"
    COMPETITION = "competition"
    REVENGE = "revenge"
    SLICE_OF_LIFE = "slice_of_life"
    OTHER = "other"


class Manga(Base):
    __tablename__ = "manga"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    normalized_title: Mapped[str] = mapped_column(String(200), unique=True, nullable=False)
    category: Mapped[MangaCategory] = mapped_column(
        Enum(MangaCategory, name="manga_category", native_enum=True, values_callable=lambda e: [m.value for m in e]),
        nullable=False,
        default=MangaCategory.OTHER,
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False
    )
